otherlibs/stdune/digest.ml

```diff
@@ -109,7 +109,9 @@ module Path_digest_result = struct
       Dune_filesystem_stubs.Unix_error.Detailed.equal x y
 end
 
-let path_with_stats ~allow_dirs path (stats : Stats_for_digest.t) :
+exception E of Dune_filesystem_stubs.Unix_error.Detailed.t
+
+let rec path_with_stats ~allow_dirs path (stats : Stats_for_digest.t) :
```

Comment:
Maybe `stats` should become an optional argument.

Reply:
Could you describe how it would help a bit more? As far as I can tell, it would only move the stat calls for directories to the beginning of the function.

Reply:
My intent was to also make the API a bit more flexible.

However, thinking about this again, making the argument optional might make it easier to forget to pass the `stats` you already have, so I retract my suggestion. Let's keep the code as is.